packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -1071,15 +1098,17 @@ export const Table = <
         >
           <ShadowTemplate>
             <slot />
-            <div
-              aria-hidden
-              style={{
-                // This ensures that the table-body is always the same width as the table-header,
-                // even if the table has no rows. See https://github.com/iTwin/iTwinUI/pull/1725
-                width: headerRef.current?.scrollWidth,
-                height: 0.1,
-              }}
-            />
+            {data.length === 0 && headerScrollWidth > headerClientWidth && (
```

Comment:
Is checking for `data` enough? What if the data is filtered?

Reply:
Ohh, right, changed (https://github.com/iTwin/iTwinUI/pull/1809/commits/f9d06f80b61bb0f2bc63adfb1b26bfc4e17a0ef6).